File: scripts/walker.lua
-- genericclient-interface: 2

local places = {
  grand_exchange = {
    label = "Grand Exchange",
    destination = { x = 3164, y = 3487, plane = 0 },
    within = 4,
  },
  varrock_center = {
    label = "Varrock Center",
    destination = { x = 3210, y = 3424, plane = 0 },
    within = 3,
  },
  edgeville_bank = {
    label = "Edgeville Bank",
    destination = { x = 3094, y = 3492, plane = 0 },
    within = 3,
  },
  falador_center = {
    label = "Falador Center",
    destination = { x = 2965, y = 3379, plane = 0 },
    within = 4,
  },
  draynor_village = {
    label = "Draynor Village",
    destination = { x = 3105, y = 3251, plane = 0 },
    within = 4,
  },
  lumbridge_castle = {
    label = "Lumbridge Castle",
    destination = { x = 3222, y = 3218, plane = 0 },
    within = 4,
  },
}

return {
  inputs = {
    {
      id = "destination",
      label = "Destination",
      type = "choice",
      default = "grand_exchange",
      choices = {
        { value = "grand_exchange", label = "Grand Exchange" },
        { value = "varrock_center", label = "Varrock Center" },
        { value = "edgeville_bank", label = "Edgeville Bank" },
        { value = "falador_center", label = "Falador Center" },
        { value = "draynor_village", label = "Draynor Village" },
        { value = "lumbridge_castle", label = "Lumbridge Castle" },
      },
    },
  },

  run = function(input)
    gc.activity("travel")
    local place = places[input.destination]
    assert(place, "Unknown Walker destination: " .. tostring(input.destination))

    gc.overlay {
      { label = "Destination", value = place.label },
      { label = "State", value = "Starting" },
    }

    gc.await { event = "game.tick" }
    gc.log("info", "walker-start", {
      place = place.label,
      destination = place.destination,
      from = gc.read("player").world,
    })

    gc.overlay {
      { label = "Destination", value = place.label },
      { label = "State", value = "Walking" },
    }

    local result = gc.await {
      action = {
        type = "walk.to",
        destination = place.destination,
        within = place.within,
      },
      timeout = { game_ticks = 600 },
    }

    if result.status == "arrived" then
      gc.overlay {
        { label = "Destination", value = place.label },
        { label = "State", value = "Arrived" },
      }
      gc.phase("travel." .. input.destination .. ".arrived")
    else
      gc.overlay {
        { label = "Destination", value = place.label },
        { label = "State", value = result.status },
      }
    end

    local idle = gc.await {
      action = { type = "mouse.offscreen" },
      breaks = false,
    }
    gc.log(result.status == "arrived" and "info" or "error", "walker-complete", {
      place = place.label,
      result = result,
      mouse = idle.status,
    })
    return result
  end,
}

File: src/main/java/com/genericclient/scripts/tools/Walker.java
package com.genericclient.scripts.tools;

import com.genericclient.script.Automation;
import com.genericclient.script.ScriptSettings;
import com.genericclient.scripts.shared.Travel;
import com.genericclient.scripts.shared.WorkflowScript;
import java.util.Map;
import org.dreambot.api.methods.map.Tile;
import org.dreambot.api.script.Category;
import org.dreambot.api.script.ScriptManifest;

@ScriptManifest(name="Walker",author="GenericClient",category=Category.UTILITY,version=1,
	description="Travel to a selected city destination.")
@ScriptSettings(id="walker",inputs=@ScriptSettings.Input(id="destination",label="Destination",
	choices={"grand_exchange","varrock_center","edgeville_bank","falador_center","draynor_village","lumbridge_castle"},
	labels={"Grand Exchange","Varrock Center","Edgeville Bank","Falador Center","Draynor Village","Lumbridge Castle"},defaultValue="grand_exchange"))
public final class Walker extends WorkflowScript
{
	private static final Map<String,Tile> PLACES = Map.of(
		"grand_exchange",new Tile(3164,3487),"varrock_center",new Tile(3210,3424),
		"edgeville_bank",new Tile(3094,3492),"falador_center",new Tile(2965,3379),
		"draynor_village",new Tile(3105,3251),"lumbridge_castle",new Tile(3222,3218));
	@Override protected Object runWorkflow()
	{
		String place = Automation.input("destination");
		Automation.overlay(Map.of("Destination",place,"State","Walking"));
		Travel.to(PLACES.get(place),3);
		Automation.phase("travel." + place + ".arrived");
		return Map.of("status","arrived","destination",place);
	}
}
